File: frontend/src/types/privacy.ts
export interface PrivacySetting {
  id: string;
  user_id: string;
  profile_visibility: "public" | "friends" | "private";
  hide_online_status: boolean;
  hide_last_seen: boolean;
  hide_birthday: boolean;
  hide_phone: boolean;
  hide_email: boolean;
  hide_work: boolean;
  hide_education: boolean;
  story_privacy: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  post_privacy: "everyone" | "friends" | "close_friends" | "followers" | "friends_followers" | "only_me" | "custom";
  reel_privacy: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  photo_privacy: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  video_privacy: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  comment_privacy: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  tag_review: boolean;
  timeline_review: boolean;
  search_engine_visibility: boolean;
  mention_permissions: "everyone" | "friends" | "close_friends" | "none";
  follow_permissions: "everyone" | "friends" | "none";
  friend_request_permissions: "everyone" | "friends" | "none";
  message_permissions: "everyone" | "friends" | "close_friends" | "none";
  call_permissions: "everyone" | "friends" | "close_friends" | "none";
  hide_friends_list: boolean;
  hide_followers_list: boolean;
  hide_following_list: boolean;
  download_media_permissions: "everyone" | "friends" | "close_friends" | "none";
  invite_permissions: "everyone" | "friends" | "close_friends" | "none";
  created_at: string;
  updated_at: string;
}

export interface PrivacySettingUpdate {
  profile_visibility?: "public" | "friends" | "private";
  hide_online_status?: boolean;
  hide_last_seen?: boolean;
  hide_birthday?: boolean;
  hide_phone?: boolean;
  hide_email?: boolean;
  hide_work?: boolean;
  hide_education?: boolean;
  story_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  post_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "friends_followers" | "only_me" | "custom";
  reel_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  photo_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  video_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  comment_privacy?: "everyone" | "friends" | "close_friends" | "followers" | "only_me" | "custom";
  tag_review?: boolean;
  timeline_review?: boolean;
  search_engine_visibility?: boolean;
  mention_permissions?: "everyone" | "friends" | "close_friends" | "none";
  follow_permissions?: "everyone" | "friends" | "none";
  friend_request_permissions?: "everyone" | "friends" | "none";
  message_permissions?: "everyone" | "friends" | "close_friends" | "none";
  call_permissions?: "everyone" | "friends" | "close_friends" | "none";
  hide_friends_list?: boolean;
  hide_followers_list?: boolean;
  hide_following_list?: boolean;
  download_media_permissions?: "everyone" | "friends" | "close_friends" | "none";
  invite_permissions?: "everyone" | "friends" | "close_friends" | "none";
}

export interface BlockedUser {
  id: string;
  full_name: string | null;
  username: string | null;
  avatar_url: string | null;
}

export interface BlockUserRequest {
  blocked_user_id: string;
}

export type BlockType = "block" | "mute" | "restrict";

File: backend/app/schemas/privacy.py
from pydantic import BaseModel, Field
from typing import Optional
from uuid import UUID
from datetime import datetime


class PrivacySettingResponse(BaseModel):
    id: UUID
    user_id: UUID
    profile_visibility: str
    hide_online_status: bool
    story_privacy: str
    post_privacy: str
    comment_privacy: str
    tag_review: bool
    timeline_review: bool
    search_engine_visibility: bool
    mention_permissions: str
    follow_permissions: str
    hide_friends_list: bool
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class PrivacySettingUpdate(BaseModel):
    profile_visibility: Optional[str] = Field(None, pattern=r"^(public|friends|private)$")
    hide_online_status: Optional[bool] = None
    story_privacy: Optional[str] = Field(None, pattern=r"^(everyone|friends|only_me|custom)$")
    post_privacy: Optional[str] = Field(None, pattern=r"^(everyone|friends|only_me|custom)$")
    comment_privacy: Optional[str] = Field(None, pattern=r"^(everyone|friends|only_me|custom)$")
    tag_review: Optional[bool] = None
    timeline_review: Optional[bool] = None
    search_engine_visibility: Optional[bool] = None
    mention_permissions: Optional[str] = Field(None, pattern=r"^(everyone|friends|only_me|none)$")
    follow_permissions: Optional[str] = Field(None, pattern=r"^(everyone|friends|none)$")
    hide_friends_list: Optional[bool] = None


class BlockUserRequest(BaseModel):
    blocked_user_id: str


class BlockUserResponse(BaseModel):
    id: UUID
    user_id: UUID
    blocked_user_id: UUID
    block_type: str
    created_at: datetime

    model_config = {"from_attributes": True}


class BlockedUserDetail(BaseModel):
    id: str
    full_name: Optional[str]
    username: Optional[str]
    avatar_url: Optional[str]

    model_config = {"from_attributes": True}
